### ISSUE TITLE
Code generated by Scrooge does not compile if a union has a member named STRING

### ISSUE THREAD

**Issue by zivanfi** [twitter/scrooge#304]
If a union has a member named STRING, the code generated for it shadows `java.lang.String`, which leads to compilation errors.

### Details

In the Apache Parquet project we would like to update our Scrooge dependecy from the ancient Scrooge 3.17.0 to a newer version. However, we have found that Scrooge 19.1.0 doesn't handle our [logical type for strings](https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L323) correctly:

```
struct StringType [...]

union LogicalType {
  1:  StringType STRING
  2: [...]
```

This leads to the following compile error (followed by several similar ones):

```
[ERROR] parquet-scrooge/target/generated-test-sources/scrooge/scrooge/org/apache/parquet/format/LogicalType.scala:510: error: type mismatch;
[ERROR]  found   : scala.collection.immutable.Map[org.apache.parquet.format.LogicalType.String,org.apache.parquet.format.LogicalType.String]
[ERROR]  required: Map[String,String]
[ERROR]         immutable$Map.empty[String, String],
```

The corresponding excerpt from the code generated by Scrooge:

```
  object String extends (StringAlias => String) {
    def withoutPassthroughFields(obj: String): String =
      withoutPassthroughFields_String(obj)

    val fieldInfo: ThriftStructFieldInfo =
      new ThriftStructFieldInfo(
        StringField,
        false,
        false,
        manifest[StringAlias],
        StringKeyTypeManifest,
        StringValueTypeManifest,
        immutable$Map.empty[String, String],
        immutable$Map.empty[String, String]
      )
  }
```

The occurances of `String` in the first line refer to our logical type, but the occurances of `String` in the `immutable$Map.empty` lines are meant to refer to `java.lang.String`. Unfortunately, the former shadows the latter, therefore the compiler interprets the unqualified `String` as `org.apache.parquet.format.LogicalType.String` instead of `java.lang.String`, leading to the compilation error.

The exact input/output files and error messages can be found below:

- [Input Thrift](https://gist.github.com/zivanfi/021c3d939fdb3660fe7a5bd9a26a8add)
- [Error Message](https://gist.github.com/zivanfi/9b7caf3fb7e2cc05324dfbd043bf051e)
- [Generated Source](https://gist.github.com/zivanfi/a812b65fb0f84dee4827c37775f30bf4)

**Comment by nandorKollar** [comment]
I propose [this](https://github.com/twitter/scrooge/pull/305) PR to fix the problem

**Comment by nandorKollar** [comment]
@zivanfi I think you can close this issue since the fixed is merged in e6ac7e1304112bb8b10f9c127a6697e65d14fbec
